pkg/node/manager/manager.go

```diff
@@ -368,7 +368,11 @@ func (m *Manager) NodeUpdated(n nodeTypes.Node) {
 		// to create symmetric traffic when sending nodeIP->pod and pod->nodeIP.
 		if address.Type == addressing.NodeCiliumInternalIP || m.conf.EncryptionEnabled() ||
 			option.Config.EnableHostFirewall || option.Config.JoinCluster {
-			tunnelIP = nodeIP
+			if tunnelIP = n.GetNodeTunnelEndpointIPv4(); tunnelIP == nil {
+				log.Debugf("No tunnelIP manually defined, just use the nodeIP")
+				tunnelIP = nodeIP
+			}
+			log.Debugf("tunnelIP %s", tunnelIP.String())
```

Comment:
nit: Even if debug is only seen by developers, if you intend to merge it into the codebase then please format the message in a way that a developer could understand it; printing random values in the debug logs makes the message difficult to interpret.

Reply:
Removed the debug log here.